main/src/ca/uwaterloo/flix/util/ExternalJarLoader.scala

```diff
@@ -29,4 +31,22 @@ class ExternalJarLoader extends URLClassLoader(Array.empty) {
     // just reimplements the superclass, but makes it public
     super.addURL(url)
   }
+
+  override def findClass(name: String): Class[_ <: Object] = {
+    try {
+      super.findClass(name)
+    } catch {
+      case e: ClassNotFoundException => 
+        // Special case for dev.flix.runtime.Global
+        // This is never used at runtime, but we need to be able to load it at compile
+        // time in order to check method signatures
+        if (name == "dev.flix.runtime.Global")
+          super.findSystemClass(name)
+        // Special case for testing to allow us to load test classes
+        else if (name.startsWith(("flix.test.")))
```

Comment:
I think this is much better than relying on the flag.

Ultimately, I would hope we could just use Javas library for tests.